scripts/check_tests_are_captured_in_ci.py

```diff
@@ -0,0 +1,269 @@
+    suite_object_string = re.sub(r'[\n\t\s]*', '', re.compile(
+        r'suites = {([^}]+)}').findall(webdriverio_config_file_content)[0])
+    test_suite_regex = re.compile(r'\b([a-zA-Z_-]*):\[(.*?)\]')
+    e2e_test_suites: List[TestSuiteDict] = []
+    for test_suite_match in test_suite_regex.findall(suite_object_string):
+        test_suite_name = test_suite_match[0]
+        test_suite_modules = [
+            test_suite for test_suite in
+            test_suite_match[1].split(',') if test_suite
+        ]
+        test_suite_modules = [
+            module.strip()[1:-1].replace('./', '')
+            for module in test_suite_modules
+        ]
+        for test_suite_module in test_suite_modules:
+            if test_suite_name in E2E_TEST_SUITES_THAT_ARE_NOT_RUN_IN_CI:
+                continue
+            e2e_test_suites.append(
+                {
+                    'name': test_suite_name,
+                    'module': test_suite_module
+                }
+            )
+    return sorted(e2e_test_suites, key=lambda x: x['name'])
+
+
+def get_acceptance_test_suites_from_acceptance_directory() -> List[TestSuiteDict]: # pylint: disable=line-too-long
+    """Gets the acceptance test suites from the acceptance test 
+    specs directory.
+    """
+
+    acceptance_test_files = glob.glob(
+        os.path.join(ACCEPTANCE_TEST_SPECS_DIRECTORY, '**/*.spec.ts'))
+    acceptance_test_suites: List[TestSuiteDict] = [
+        {
+            'name': os.path.relpath(
+                module,
+                ACCEPTANCE_TEST_SPECS_DIRECTORY
+            ).replace('.spec.ts', ''),
+            'module': os.path.relpath(module, os.getcwd())
+        } for module in acceptance_test_files
+    ]
+    return sorted(acceptance_test_suites, key=lambda x: x['name'])
+
+
+def compute_test_suites_difference(
+    test_suites_from_config: List[TestSuiteDict],
+    test_suites_from_directory: List[TestSuiteDict]
+) -> List[TestSuiteDict]:
+    """Computes the difference between the test suites from the CI config files
+    and the test suites from the test files.
+
+    Args:
+        test_suites_from_config: list(dict). A list of dictionaries sorted
+            by name, each containing the name and module of a test suite from
+            the CI config files.
+        test_suites_from_directory: list(dict). A list of dictionaries sorted 
+            by name, each containing the name and module of a test suite from
+            the test files.
+
+    Returns:
+        list(dict). A list of test suites which are not present in both 
+        the CI config files and the test files, but are present in
+        either of them.
+    """
+    test_suites_difference = [
+        test_suite for test_suite in test_suites_from_directory +
+        test_suites_from_config if test_suite not in test_suites_from_config or
+        test_suite not in test_suites_from_directory
+    ]
+    return test_suites_difference
```

Comment:
Can you use sets to do this?

Reply:
I think I did this similar to not using sets above is because dictionaries are unhashable, and sets are like dictionaries, so you can't have a key being a dictionary.